components/invoice.js

```diff
@@ -238,12 +231,13 @@ export const useInvoiceable = (fn, options = defaultOptions) => {
         throw error
       }
     }
-    setFnArgs(args)
-    await createInvoice({ variables: { amount } })
+    setFormValues(formValues)
+    setSubmitArgs(submitArgs)
+    await createInvoice({ variables: { amount: cost } })
     // tell onSubmit handler that we want to keep local storage
     // even though the submit handler was "successful"
     return { keepLocalStorage: true }
```

Comment:
do we still need this now that Form is invoicable-aware?

(This was one of the things that tipped us off that original pattern might be improved.)

Reply:
Mhh, yes, that was one of the things that tipped of us off to improve the original pattern. But I realized that moving the error handling code into the form code wouldn't be enough. For example, at this line here, it's not about errors but communicating back on success when the form should clear local storage or not.

So I am not sure how to fit that logic into the form code. It would have to be able to distinguish between different submit handlers successes which it can't since we just pass an opaque `onSubmit` to it and then additionally (and conditionally) wrap it with `useInvoiceable`. If this wrapped function is called successfully, form can not know by itself if it should clear local storage since the action was submitted successfully or if only an invoice was created and the invoice modal opened afaict

Reply:
Cool sounds good. I still haven't pulled to take a close look. I'll let you know if any simplifications jump out.